### PR TITLE
support utf-8 files containing BOM

### PR DESCRIPTION
**Pull request by tolry**
currently accessing the first column in a file containing a byte order mark does not work properly

### CODE REVIEW

**Comment by tolry** [comment]
Might be useful to have a test for this, right? Should I start a test suite?

**Comment by DavidBadura** [comment]
Yes, this would be perfect :)

File: src/CsvParser.php [hunk 1]
```diff
@@ -8,6 +8,10 @@
  */
 class CsvParser implements \Iterator
 {
+    /**
+     * @var string
+     */
+    static private $bom = "\xef\xbb\xbf";
```
Comment (DavidBadura) [inline src/CsvParser.php]: why you don't use a constant here?

Comment (tolry) [inline src/CsvParser.php]: imo, this is an internal technical detail and should therefore not be public - constants cannot be private/protected in php :disappointed:

Comment (DavidBadura) [inline src/CsvParser.php]: ah, you are right. :+1:

**Comment by tolry** [comment]
@DavidBadura take a look - just that single test, could probably be more, but it's a start :smile:  how about travis?

**Comment by DavidBadura** [comment]
Thx, I will do this soon.

**Comment by tolry** [comment]
Would you mind tagging the current version?

**Comment by DavidBadura** [comment]
done :wink:

**Comment by tolry** [comment]
thx

**Pull request merged**